app/services/db/establish_connection/read_replica_connection.rb

```diff
@@ -0,0 +1,39 @@
+module Db
+  module EstablishConnection
+    class ReadReplicaConnection
+      def call
+        return yield if Rails.env.test?
```

Comment:
It kind of bugs me that `yield` is there twice (because it's side-effecty) but I can't think of anything better -- I see it's guarded by the `return` so I know it's fine.

Reply:
Only alternative I can think of is adding like a `if !Rails.env.test` to the end of the next line but that's sneaky and confusing